electron-app/magnifier/rust-sampler-manager.ts

```
@@ -0,0 +1,273 @@
+            resolve();
+          }
+        },
+        (error) => {
+          if (!resolved) {
+            resolved = true;
+            clearTimeout(timeout);
+            reject(new Error(error));
+          }
+        }
+      );
+    });
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)